app/src/components/code/JSONToolEditor.tsx

```diff
@@ -0,0 +1,47 @@
+import React from "react";
+import { json, jsonLanguage, jsonParseLinter } from "@codemirror/lang-json";
+import { linter } from "@codemirror/lint";
+import { EditorView, hoverTooltip } from "@codemirror/view";
+import { githubLight } from "@uiw/codemirror-theme-github";
+import { nord } from "@uiw/codemirror-theme-nord";
+import CodeMirror, { ReactCodeMirrorProps } from "@uiw/react-codemirror";
+import {
+  handleRefresh,
+  jsonCompletion,
+  jsonSchemaHover,
+  jsonSchemaLinter,
+  stateExtensions,
+} from "codemirror-json-schema";
+import { JSONSchema7 } from "json-schema";
+
+import { useTheme } from "@phoenix/contexts";
+import { toolJSONSchema } from "@phoenix/schemas/toolSchema";
+
+export type JSONToolEditorProps = Omit<
+  ReactCodeMirrorProps,
+  "theme" | "extensions" | "editable"
+>;
+
+export function JSONToolEditor(props: JSONToolEditorProps) {
```

Comment:
Isn't this really JSONEditor with json schema added on top?

Reply:
Yeah i had baked in the tool schema, so that's what makes it "tool" I guess but i could either
- modify the json editor to accept a schema and conditionally add the extensions or
- make a new jsoneditorwithschema (still would need to take in a schema as a prop)

Reply:
removed and extended the jsonEditor